operation/manager_test.go

```diff
@@ -97,22 +97,39 @@ func TestDontCancel(t *testing.T) {
 }
 
 func TestCancelRace(t *testing.T) {
+	// First set up the "worker" context and register an operation on
+	// the `SingleOperationManager` instance.
 	som := SingleOperationManager{}
-	ctx, done := som.New(context.Background())
-	defer done()
-
-	var wg sync.WaitGroup
-
-	wg.Add(1)
-	go func() {
-		_, done := som.New(context.Background())
-		wg.Done()
-		defer done()
-	}()
+	workerError := make(chan error, 1)
+	workerCtx, somCleanupFunc := som.New(context.Background())
+
+	// Spin up a separate go-routine for the worker to listen for cancelation. Canceling an
+	// operation blocks until the operation completes. This goroutine is responsible for running
+	// `somCleanupFunc` to signal that canceling has completed.
+	workerFunc := func() {
+		defer somCleanupFunc()
+
+		select {
+		case <-workerCtx.Done():
+			workerError <- nil
+		case <-time.After(5 * time.Second):
+			workerError <- errors.New("Failed to be signaled via a cancel")
+		}
 
-	som.CancelRunning(ctx)
-	wg.Wait()
-	test.That(t, ctx.Err(), test.ShouldNotBeNil)
+		close(workerError)
+	}
+	go workerFunc()
+
+	// Set up a "test" context to cancel the worker.
+	testCtx, testCleanupFunc := context.WithTimeout(context.Background(), time.Second)
+	defer testCleanupFunc()
+	som.CancelRunning(testCtx)
+	// If `workerCtx.Done` was observed to be closed, the worker thread will pass a `nil` error back.
+	test.That(t, <-workerError, test.ShouldBeNil)
+	// When `SingleOperationManager` cancels an operation, the operation's context should contain a
+	// "context canceled" error message.
```

Comment:
(for the next patch) I've edited this comment to reflect we don't check the error message but the error object itself as per line 132.